internal/engine/rtengine/cache.go

```diff
@@ -121,6 +134,21 @@ func cacheRuleEngine(
 		return nil, fmt.Errorf("error parsing rule type when parsing rule type %s: %w", ruleType.ID, err)
 	}
 
+	// Build a registry instance per rule type. This allows us to have an
+	// isolated data source list per instance of the rule type engine which is
+	// what we want. We don't want rule types using data sources they haven't
+	// instantiated. It is in this spot that we would add something like a cache
+	// so data sources could optimize in a per-execution context.
+	//
+	// TODO: Do we need to pass in a transaction here?
+	// TODO: We _might_ want to pass in a slice of the hierarchy here.
+	dsreg, err := dssvc.BuildDataSourceRegistry(ctx, pbRuleType, nil)
```

Comment:
It's not possible for this to conflict right? I.e. one ruletype having access to the data sources for another one?

Reply:
That's right. This is why I make the instantiation of the registry here and per rule type. The registry is not shared.

Reply:
Perfect! 💯 😃